src/finalfusion/compat/word2vec.py

```diff
@@ -0,0 +1,109 @@
+"""
+Word2vec binary format.
+"""
+
+import sys
+from os import PathLike
+from typing import Union, BinaryIO, AnyStr
+
+import numpy as np
+
+from finalfusion import Embeddings
+from finalfusion.io import _serialize_array_as_le
+from finalfusion.storage import NdArray
+from finalfusion._util import _normalize_ndarray_storage
+from finalfusion.vocab import SimpleVocab
+
+
+def load_word2vec(file: Union[str, bytes, int, PathLike]) -> Embeddings:
+    """
+    Read embeddings in word2vec binary format.
+
+    The returned embeddings have a SimpleVocab, NdArray storage and a Norms chunk. The storage is
+    l2-normalized per default and the corresponding norms are stored in the Norms.
+
+    Files are expected to start with a line containing rows and cols in utf-8. Words are encoded
+    in utf-8 followed by a single whitespace. After the whitespace, the embedding components are
+    expected as little-endian single-precision floats.
+
+    Parameters
+    ----------
+    file : str, bytes, int, PathLike
+        Path to a file with embeddings in word2vec binary format.
+
+    Returns
+    -------
+    embeddings : Embeddings
+        The embeddings from the input file.
+    """
+    words = []
+    with open(file, 'rb') as inf:
+        rows, cols = map(int, inf.readline().decode("utf-8").split())
```

Comment:
Do you expect UTF-8? ;)

Reply:
#19 I guess ;) Did we decide that finalfusion files always contain well-formed UTF-8? Forgot if we specified that in the spec...

Reply:
I meant more that the dimensions are supposed to be ASCII.

Who knows what happens when you plug in unicode subscript/superscript numerals such as ₁¹. I think it's better to decode the dimensions as ASCII (not latin-1), so that decoding fails if someone put something weird as the dimensionality.